doc/reference/vm_live_migration_internals.md

```diff
@@ -0,0 +1,127 @@
+
+API extension: `migration_vm_live`
+
+## Conceptual process
+
+The live migration workflow varies depending on the type of storage pool used. The two key scenarios are non-shared storage and shared storage within a cluster (e.g., Ceph). If live state transfer is not supported by a target, a stateful stop is performed prior to migration.
+
+### Live migration for non-shared storage
+
+This process leverages the QEMU built-in Network Block Device (NBD) client-server mechanism to transfer the virtual machine's disk and memory state. Below is an overview of the steps:
+
+1. Set up connection.
+1. Determine migration type (shared or non-shared storage).
+1. Set migration capabilities.
+1. Non-shared storage preparation.
+- Create and configure snapshot file for root disk writes during migration;
+- Add the snapshot as a block device to the source VM;
+- Redirect disk writes to the snapshot.
+1. Storage transfer.
+- For shared storage, we just perform checks at this point;
+- For non-shared storage, set up an NBD listener and connect it to the target to transfer the disk.
+1. Snapshot sync for non-shared storage to ensure consistency between source and target.
+1. Transfer VM state to target.
+
+```{figure} /images/vm_live_migration_flowchart.svg
+Non-Shared Storage Migration
+```
+
+The state transitions during the process are shown below:
+
+```{figure} /images/vm_live_migration_fsa_diagram.svg
+Non-Shared Storage Migration State Transitions
+```
+
+### Intra-cluster member live migration (Ceph shared storage pool)
+
+For shared storage pools such as Ceph, disk data transfer is unnecessary. Instead, the process focuses on transferring the VM state through a dedicated migration socket:
+
+1. Validate cluster state and storage pool readiness.
+1. Notify the shared disks that they will be accessed from another system.
+1. Pause the guest OS on the source VM and transfer the live state data over the migration socket.
+1. Stop and delete source VM.
+1. Start the target VM using the transferred state.
+
+
+## Migration API
+
+Sending a `POST` request to `/1.0/instances/{name}` renames, moves an instance between pools, or migrates an instance to another server. In the push case, the returned operation metadata for migration is a background operation with progress data. For the pull case, it is a WebSocket operation with a number of secrets to be passed to the target server.
+
+## Live migration call stack
+
+Below is a general overview of the key functions of the live migration call stack:
+
+### [`lxd/lxd/instance_post.go`](https://github.com/canonical/lxd/blob/main/lxd/instance_post.go)
+
+1. [`instancePost`](https://github.com/canonical/lxd/blob/main/lxd/instance_post.go#L74)
+
+This function handles post requests to the `/1.0/instances` endpoint.
+
+### [`lxd/lxd/migrate_instance.go`](https://github.com/canonical/lxd/blob/main/lxd/migrate_instance.go)
+
+1. [`Do`](https://github.com/canonical/lxd/blob/main/lxd/migrate_instance.go#L87)
```

Comment:
"Do"? Is there supposed to be more text here?

Reply:
No, the function is literally named `Do`.